src/app/public/modules/select-field/select-field.component.ts

```diff
@@ -185,6 +192,8 @@ export class SkySelectFieldComponent implements ControlValueAccessor, OnDestroy
         pickerContext.data = this.data;
         pickerContext.selectedValue = this.value;
         pickerContext.selectMode = this.selectMode;
+        pickerContext.showNewButton = this.showNewButton;
+        pickerContext.newButtonEvent = this.newButtonEvent;
```

Comment:
Instead of passing the event handler directly to the modal context, perhaps the context instance could have its own observable that, when fired, would call the `addNewRecordButtonClick()` event from within the select-field component. This will protect the scope of the `SkySelectFieldComponent` a little better.

**Add a new observable to select-field-picker-context.ts**
```
export class SkySelectFieldPickerContext {

  public get addNewButtonClick(): Observable {
    return this._addNewButtonClick;
  }

  private _addNewButtonClick = new Subject<void>();

  // ... ... ...
}
```

**The select-field.component.ts subscribes to the context's observable**
```
pickerContext.addNewButtonClick.subscribe(() => {
  this.addNewButtonClick.next();
});
```

**select-field-picker.component.ts**
```
public onAddNewButtonClick(): void {
  this.context.addNewButtonClick.next();
}
```

Reply:
How will the select-field-picker  this.context.addNewButtonClick.next(); get next from  public get addNewButtonClick(): Observable